### PR TITLE
Introduce CSSPixel as a replacement for ViewportPx and PagePx.

### PR DESCRIPTION
**Pull request by glennw**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15700)

### CODE REVIEW

**Comment by glennw** [comment]
This change deserves some explanation.

This PR is (sort of) a WIP. I want to get some feedback on whether
this seems like a viable / correct approach to handling both
page and pinch zoom. Assuming it does, we can either merge this
after review (which would be my preference) or wait until the WR
pinch zoom functionality is ready before merging this part.

The idea is to move pinch-zoom to be handled by WR as (almost)
an internal detail. There would be an API to control the pinch zoom
factor, but apart from that WR would handle pinch-zoom internally.

From what I can tell of the specifications and browser behaviour,
this makes sense - but it's quite possible I've missed an important
detail.

Given the assumptions above, the Servo code now only needs to deal
with three coordinate spaces, rather than four. Specifically, we now
would have:
    Device pixels
        -> Device independent pixels (hi-dpi factor)
            -> CSS pixels (page zoom factor)

This PR doesn't break any existing functionality, since pinch zoom
doesn't currently work at all. Page zoom is also currently broken
although that's orthogonal to this PR.

**Comment by glennw** [comment]
Rebased and fixed tidy error.

**Comment by jdm** [comment]
Style unit tests need updating.

**Comment by glennw** [comment]
Thanks, updated the unit tests.

File: components/style_traits/lib.rs [hunk 1]
```diff
@@ -26,33 +26,41 @@ extern crate rustc_serialize;
 /// Must be transmutable to and from `TNode`.
 pub type UnsafeNode = (usize, usize);
 
+/// Represents a mobile style pinch zoom factor.
+/// TODO(gw): Once WR supports pinch zoom, use a type directly from webrender_traits.
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(feature = "servo", derive(Deserialize, Serialize, HeapSizeOf))]
+pub struct PinchZoomFactor(f32);
```
Comment (mbrubeck) [inline components/style_traits/lib.rs]: Why a new type instead of ScaleFactor?

**Comment by mbrubeck** [comment]
@bors-servo r+

Curious about the answer to the above question, but no changes required.

**Pull request merged**